Flow.Launcher.Core/Plugin/PluginsLoader.cs

```diff
@@ -129,47 +133,104 @@ public static IEnumerable<PluginPair> PythonPlugins(List<PluginMetadata> source,
-                }
             }
             else
             {
-                var path = Path.Combine(pythonDirectory, PythonExecutable);
+                var path = Path.Combine(settings.PythonDirectory, PythonExecutable);
                 if (File.Exists(path))
                 {
                     Constant.PythonPath = path;
                 }
                 else
                 {
-                    Log.Error($"|PluginsLoader.PythonPlugins|Can't find python executable in {path}");
+                    Log.Error("PluginsLoader",$"Tried to automatically set from Settings.PythonDirectory " +
+                        $"but can't find python executable in {path}", "PythonPlugins");
                 }
             }
 
-            // if we have a path to the python executable,
-            // load every python plugin pair.
-            if (String.IsNullOrEmpty(Constant.PythonPath))
+            if (string.IsNullOrEmpty(settings.PythonDirectory))
             {
-                return new List<PluginPair>();
+                if (MessageBox.Show("Flow detected you have installed Python plugins, " +
+                        "would you like to install Python to run them? " +
+                        Environment.NewLine + Environment.NewLine +
+                        "Click no if it's already installed, " +
+                        "and you will be prompted to select the folder that contains the Python executable",
+                        string.Empty, MessageBoxButtons.YesNo) == DialogResult.No
+                    && string.IsNullOrEmpty(settings.PythonDirectory))
+                {
+                    var dlg = new FolderBrowserDialog
+                    {
+                        SelectedPath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles)
+                    };
+
+                    var result = dlg.ShowDialog();
+                    if (result == DialogResult.OK)
+                    {
+                        string pythonDirectory = dlg.SelectedPath;
+                        if (!string.IsNullOrEmpty(pythonDirectory))
+                        {
+                            var pythonPath = Path.Combine(pythonDirectory, PythonExecutable);
+                            if (File.Exists(pythonPath))
+                            {
+                                settings.PythonDirectory = pythonDirectory;
+                                Constant.PythonPath = pythonPath;
+                            }
+                            else
+                            {
+                                MessageBox.Show("Can't find python in given directory");
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    var installation = Task.Run(async () => { await DroplexPackage.Drop(App.python3_9_1).ConfigureAwait(false); });
+                    installation.Wait();
```

Comment:
emmm since we are waiting it, why do we need to wrap it within the Task.Run?

Reply:
@taooceros this should be sychronous, is there another way to do this without converting the calling methods async?

Reply:
You can simply Wait the task do be done. `DroplexPackage.Drop(App.python3_9_1).Wait(timeout)`, while timeout is the time to be waited. However, actually, I don't think it would be great to let user wait downloading python at initialization, since this will block other Flow's functionality.
I think this should be done async, disabling python plugin first if python not installed (also ask whether user has installed python similar to what has done in Everything plugin), and once it has downloaded successfully, enabling them.

Reply:
I think it's fine because the installation has a prompt from Python, so user knows it's installing.

It does also ask the user if they want to select the location beforehand as well

Reply:
oh ok I haven't looked at the newer part of the code.
However, I still think we should wait python installation at initialization, but let it does the job on background.
Is it possible to disable all python plugin and then later enable them after python installed?

Reply:
yeah enabling and disabling is what i dont want to mess with at this stage. If we get any issues from users that it's taking forever to download then we can look to do that, the install is not big and should download and install quickly.

Reply:
I don't think python includes a great cdn so users in China may encounter significant of time of downloading it. However, I do agree with you that disabling and enabling plugins will bring quite a mess with at this stage.
Maybe don't load python plugin, and force restarting Flow after Python has installed?

Reply:
python install file is only 28mbs, can you test it out see if it takes long

Reply:
> python install file is only 28mbs, can you test it out see if it takes long

Will take a try.

Reply:
hmmm seems that people in China cannot directly finish download python because of timeout. I have encountered that the task was cancelled when I try to do this.
![image](https://user-images.githubusercontent.com/45326534/107144786-6ebffd00-6978-11eb-9499-8d8610baec6c.png)

Reply:
Thank you for testing. How long before it's timeout? 

Is it timing out on connection or during the downloading?